WordPress/Classes/Utility/ImmuTable.swift

```diff
@@ -320,47 +322,47 @@ open class ImmuTableViewHandler: NSObject, UITableViewDataSource, UITableViewDel
 
     open func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
         if target.responds(to: #selector(UITableViewDelegate.tableView(_:heightForFooterInSection:))) {
-            return target.tableView(tableView, heightForFooterInSection: section)
+            return target.tableView?(tableView, heightForFooterInSection: section) ?? UITableView.automaticDimension
         }
 
         return UITableView.automaticDimension
     }
 
     open func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
         if target.responds(to: #selector(UITableViewDelegate.tableView(_:heightForHeaderInSection:))) {
-            return target.tableView(tableView, heightForHeaderInSection: section)
+            return target.tableView?(tableView, heightForHeaderInSection: section) ?? UITableView.automaticDimension
         }
 
         return UITableView.automaticDimension
     }
 
     open func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
         if target.responds(to: #selector(UITableViewDelegate.tableView(_:viewForFooterInSection:))) {
-            return target.tableView(tableView, viewForFooterInSection: section)
+            return target.tableView?(tableView, viewForFooterInSection: section)
         }
 
         return nil
     }
 
     open func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
         if target.responds(to: #selector(UITableViewDelegate.tableView(_:viewForHeaderInSection:))) {
-            return target.tableView(tableView, viewForHeaderInSection: section)
+            return target.tableView?(tableView, viewForHeaderInSection: section)
         }
 
         return nil
     }
 
     open func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
         if target.responds(to: #selector(UITableViewDataSource.tableView(_:canEditRowAt:))) {
-            return target.tableView(tableView, canEditRowAt: indexPath)
+            return target.tableView?(tableView, canEditRowAt: indexPath) ?? false
         }
 
         return false
     }
 
     open func tableView(_ tableView: UITableView, editActionsForRowAt indexPath: IndexPath) -> [UITableViewRowAction]? {
         if target.responds(to: #selector(UITableViewDelegate.tableView(_:editActionsForRowAt:))) {
-            return target.tableView(tableView, editActionsForRowAt: indexPath)
+            return target.tableView?(tableView, editActionsForRowAt: indexPath)
```

Comment:
Because of this line, we need the target to conform to `UITableViewDataSource`. I'd love to remove this (since THIS class is the Data Source), but I couldn't come up with anything :(